installer/pkg/components/server/ide-configmap.go

```diff
@@ -14,32 +14,101 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
 )
 
-func CodeImageStableVersion(ctx *common.RenderContext) string {
-	stableVersion := ctx.VersionManifest.Components.Workspace.CodeImageStable.Version
-	if stableVersion == "" {
-		stableVersion = ctx.VersionManifest.Components.Workspace.CodeImage.Version
-	}
-	return stableVersion
-}
-
 func ideconfigmap(ctx *common.RenderContext) ([]runtime.Object, error) {
-	stableVersion := CodeImageStableVersion(ctx)
+	typeBrowser := "browser"
+	typeDesktop := "desktop"
 	idecfg := IDEConfig{
-		IDEVersion:   stableVersion,
-		IDEImageRepo: workspace.CodeIDEImage,
+		IDEVersion:   workspace.CodeIDEImageStableVersion,
+		IDEImageRepo: common.RepoName(ctx.Config.Repository, workspace.CodeIDEImage),
 		IDEImageAliases: map[string]string{
-			"code":        common.ImageName(ctx.Config.Repository, workspace.CodeIDEImage, stableVersion),
+			"code":        common.ImageName(ctx.Config.Repository, workspace.CodeIDEImage, workspace.CodeIDEImageStableVersion),
 			"code-latest": common.ImageName(ctx.Config.Repository, workspace.CodeIDEImage, ctx.VersionManifest.Components.Workspace.CodeImage.Version),
 		},
 		DesktopIDEImageAliases: map[string]string{
 			"code-desktop":          common.ImageName(ctx.Config.Repository, workspace.CodeDesktopIDEImage, ctx.VersionManifest.Components.Workspace.DesktopIdeImages.CodeDesktopImage.Version),
 			"code-desktop-insiders": common.ImageName(ctx.Config.Repository, workspace.CodeDesktopInsidersIDEImage, ctx.VersionManifest.Components.Workspace.DesktopIdeImages.CodeDesktopImageInsiders.Version),
-			"intellij:":             common.ImageName(ctx.Config.Repository, workspace.IntelliJDesktopIDEImage, ctx.VersionManifest.Components.Workspace.DesktopIdeImages.IntelliJImage.Version),
+			"intellij":              common.ImageName(ctx.Config.Repository, workspace.IntelliJDesktopIDEImage, ctx.VersionManifest.Components.Workspace.DesktopIdeImages.IntelliJImage.Version),
 			"goland":                common.ImageName(ctx.Config.Repository, workspace.GoLandDesktopIdeImage, ctx.VersionManifest.Components.Workspace.DesktopIdeImages.GoLandImage.Version),
 		},
 		SupervisorImage: common.ImageName(ctx.Config.Repository, workspace.SupervisorImage, ctx.VersionManifest.Components.Workspace.Supervisor.Version),
+		IDEOptions: IDEOptions{
+			Options: map[string]IDEOption{
+				"theia": {
+					Title:   "Theia (legacy)",
+					Tooltip: pointer.String("This entry exists solely for legacy reasons."),
```

Comment:
This file is auto-generated, right?

Reply:
No, that's a duplicate of the content of the configmap helm template. As far as I understood it, this part of the installer will replace the configmap template and will be the single point of truth eventually.